cryptofeed/pairs.py

```diff
@@ -40,46 +41,60 @@ def gen_pairs(exchange: str, key_id=None):
 def _binance_pairs(endpoint: str, exchange: str):
     ret = {}
     pairs = requests.get(endpoint).json()
-    for symbol in pairs['symbols']:
-        split = len(symbol['baseAsset'])
-        normalized = symbol['symbol'][:split] + PAIR_SEP + symbol['symbol'][split:]
-        ret[normalized] = symbol['symbol']
-        _exchange_info[exchange]['tick_size'][normalized] = symbol['filters'][0]['tickSize']
-        if "contractType" in symbol:
-            _exchange_info[exchange]['contract_type'] = symbol['contractType']
+    try:
+        for symbol in pairs['symbols']:
+            split = len(symbol['baseAsset'])
+            normalized = symbol['symbol'][:split] + PAIR_SEP + symbol['symbol'][split:]
+            ret[normalized] = symbol['symbol']
+            _exchange_info[exchange]['tick_size'][normalized] = symbol['filters'][0]['tickSize']
+            if "contractType" in symbol:
+                _exchange_info[exchange]['contract_type'] = symbol['contractType']
+    except KeyError:
+        LOG.critical("GET BINANCE pairs %s: Unexpected response: %r", exchange, pairs)
+        raise
     return ret
 
 
-def binance_pairs():
+def binance_pairs() -> Dict[str, str]:
     return _binance_pairs('https://api.binance.com/api/v1/exchangeInfo', BINANCE)
 
 
-def binance_us_pairs():
+def binance_us_pairs() -> Dict[str, str]:
     return _binance_pairs('https://api.binance.us/api/v1/exchangeInfo', BINANCE_US)
 
 
-def binance_futures_pairs():
+def binance_futures_pairs() -> Dict[str, str]:
     return _binance_pairs('https://fapi.binance.com/fapi/v1/exchangeInfo', BINANCE_FUTURES)
 
 
-def binance_delivery_pairs():
+def binance_delivery_pairs() -> Dict[str, str]:
     return _binance_pairs('https://dapi.binance.com/dapi/v1/exchangeInfo', BINANCE_DELIVERY)
 
 
-def bitfinex_pairs():
+def bitfinex_pairs() -> Dict[str, str]:
+    # doc: https://docs.bitfinex.com/docs/ws-general#supported-pairs
+    tickers: List[List[str]] = requests.get("https://api.bitfinex.com/v2/tickers?symbols=ALL").json()
+    norm: List[List[str]] = requests.get("https://api-pub.bitfinex.com/v2/conf/pub:map:currency:sym").json()[0]
+    norm: Dict[str, str] = dict(norm)
+    for k, v in dict(norm).items():
+        if k[-2:] == "F0" or '-' in v:  # Do not convert BTCF0 -> BTC or PBTCETH -> PBTC-ETH
+            del norm[k]
     ret = {}
-    r = requests.get('https://api.bitfinex.com/v2/tickers?symbols=ALL').json()
-    for data in r:
-        pair = data[0]
+    for pair in [t[0] for t in tickers]:
         if pair[0] == 'f':
-            continue
-        normalized = pair[1:-3] + PAIR_SEP + pair[-3:]
-        normalized = normalized.replace('UST', 'USDT')
```

Comment:
Hi @olibre 
Please, keep this normalization step. Bitfinex knows `USDT` as `UST`.

Also, could you add a line making the same to replace `BCHN`as `BCH`?
Most of the main other exchanges have kept `BCH` for Bitcoin Cash Node (Kraken, Huobi & Binance most notably)

Thanks

Reply:
it converts UST to USDt. I added a line that makes it USDT. I'll add something to force BCHN -> BCH

Reply:
done